content/en/docs/concepts/extend-kubernetes/compute-storage-net/resource-drivers.md

```diff
@@ -0,0 +1,233 @@
+{{< glossary_tooltip term_id="resource-claim" text="ResourceClaims">}}.
+
+There are two modes of allocation the ResourceClaim can have:
+
+- `WaitForFirstConsumer` (default), which you could think of as meaning _delayed_.
+  In this mode the cluster only requests resource(s)
+  for ResourceClaim when a Pod that needs it is being scheduled.
+- `Immediate`: the resource has to be allocated to the ResourceClaim as soon as possible, and
+  retained until the ResourceClaim is deleted.
+
+If `allocationMode` is not set explicitly, the mode is `WaitForFirstConsumer`.
+
+### Delayed allocation
+
+Delayed allocation is done in two steps: first, the scheduler picks up a list of potential Node names
+and asks from resource driver controller which of them are unsuitable for ResourceClaims through
+_UnsuitableNodes_ call. If no nodes were suitable, scheduler selects another batch of Node names for
+the same call, until suitable node is found.
+
+When at least one Node is found to be suitable for all ResourceClaims, the scheduler
+considers the suitable nodes for the other Pod scheduling constraints (native resources
+requests, affinity, selectors, etc.), and picks up exactly one Node name. The resource driver
+controller is then asked to allocate resources for the ResourceClaim on the selected node through
+_Allocate_ call. All the ResourceClaims referenced by the Pod that match the resource driver name
+in the ResourceClass, will be passed as parameters into this call.
+
+If Allocate call returns error for any number of ResourceClaims, the helper code will repeat the
+same call with interval until it succeeds.
+
+### Immediate allocation
+
+Immediate allocation does not have selected node, and it is up to the resource driver controller
+to select the best suitable node based on the ResourceClaim, ResourceClass and their parameters.
+Therefore in this scenario only `Allocate` is called (without `UnsupportedNodes` being called first).
+
+### Common calls for both allocation modes
+
+In both modes the allocation is preceded by getting parameters objects for ResourceClaims and
+ResourceClasses to ensure the resource driver is able to get these objects and understand them.
+
+## Sharing resources
+
+There are two main ways of sharing the resources between Pods:
+- by using the same ResourceClaim in multiple Pods
+- by allocating the same resource to multiple ResourceClaims
+
+### Shared ResourceClaims
+
+If the `Shareable` field is set to `true` in AllocationResult for ResourceClaim, scheduler will
+allow the same ResourceClaim to be used by up to 32 Pods by automatically updating
+`Claim.Status.ReservedFor` field without consulting the resource driver that allocated resource
+for this ResourceClaim.
+
+This means that the same node resource can be used by multiple applications, which potentially can lead to service degradation and therefore needs to be considered carefully when AllocationResult is
+written.
```

Comment:
This is not something that the driver or vendor need to decide. The driver just communicates that it's technically feasible to share a claim between pods.

Whether that makes sense is something that the user needs to decide based on the application that they deploy. Just because a claim is sharable does not mean that it automagically will be shared - the application needs to be deployed such that it does this.

Reply:
Removed last paragraph.